src/OpenApi/test/Extensions/TypeExtensionsTests.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.AspNetCore.OpenApi;
+
+public class TypeExtensionsTests
+{
+    private delegate void TestDelegate(int x, int y);
+
+    public static IEnumerable<object[]> GetSchemaReferenceId_Data =>
+    [
+        [typeof(Todo), "Todo"],
+        [typeof(IEnumerable<Todo>), "TodoIEnumerable"],
+        [typeof(TodoWithDueDate), "TodoWithDueDate"],
+        [typeof(IEnumerable<TodoWithDueDate>), "TodoWithDueDateIEnumerable"],
+        [(new { Id = 1 }).GetType(), "Int32AnonymousType"],
+        [(new { Id = 1, Name = "Todo" }).GetType(), "Int32StringAnonymousType"],
+        [typeof(IFormFile), "IFormFile"],
+        [typeof(IFormFileCollection), "IFormFileCollection"],
+        [typeof(Results<Ok<TodoWithDueDate>, Ok<Todo>>), "TodoWithDueDateOkTodoOkResults"],
```

Comment:
If this is supposed to be human-readable, some underscores might help.

Reply:
It's not defined in the spec. The only constraints that the spec puts on these keys is that they have to be case sensitive.

I opted for alphanumeric values (avoiding things like "<" or "`").

> If this is supposed to be human-readable, some underscores might help.

Adding this would help although underscores as separators isn't a convention I've usually seen. Conventionally, most APIs tend to use [PascalCase](https://github.com/OAI/OpenAPI-Specification/blob/main/examples/v3.0/petstore-expanded.json#L210) for type names.

Reply:
I meant between types, not within types.  TodoWithDueDateOk_TodoOk_Results

Reply:
That would be `TodoWithDueDate_Ok_Todo_Ok_Results` in this case?
Throwing out random thought: `ResultsOf_OkOf_TodoWithDueDate_And_OkOf_Todo`

Reply:
There's enough context that the printer could skip it for arity 1, but sure, might as well go all the way.  I like "Of", but I find "And" pretty wordy.